test/unit/extension/lib/getAuthenticatedCheckoutUrlTest.js

```diff
@@ -0,0 +1,38 @@
+const assert = require('assert')
+
+const getAuthenticatedCheckoutUrl = require('../../../../extension/lib/getAuthenticatedCheckoutUrl')
+
+describe('getAuthenticatedCheckoutUrl - unit', () => {
+  const baseUrl = 'https://my-shop.com'
+  const context = {
+    config: {
+      storeHash: 'storehash',
+      accessToken: 'accesstoken',
+      clientId: 'clientid',
+      clientSecret: 'clientSecret',
+      requestTimeout: '15000'
+    },
+    log: {
+      error: msg => {}
+    }
+  }
+  const input = {
+    url: baseUrl + '/cart.php?id=123456'
+  }
+
+  it('should return token url', async () => {
+    const meta = {
+      userId: '15358639'
+    }
+    const response = await getAuthenticatedCheckoutUrl({...context, meta}, input)
+    assert.equal(response.url.indexOf(baseUrl + '/login/token/'), 0)
```

Comment:
I am not sure how to stub or mock the AuthRepository in a way that would cause the getAuthenticatedCheckoutUrl would use it because the AuthRepository is not passed into the getAuthenticatedCheckoutUrl method. Can you show me how to do this?